test/Runtime/demangleToMetadata.swift

```diff
@@ -245,11 +245,17 @@ struct ConformsToP1: P1 { }
 struct ConformsToP2: P2 { }
 struct ConformsToP3: P3 { }
```

Comment:
Can you also test type reconstruction with nested structs and aliases? In test/TypeDecoder/